woocommerce/payment-gateway/class-sv-wc-payment-gateway-payment-form.php

```diff
@@ -99,7 +102,23 @@ protected function add_hooks() {
 		add_action( "wc_{$gateway_id}_payment_form_end",   array( $this, 'render_fieldset_end' ), 5 );
 
 		// payment form JS
-		add_action( "wc_{$gateway_id}_payment_form_end",   array( $this, 'render_js' ), 5 );
+		add_action( "wc_{$gateway_id}_payment_form_end", [ $this, 'render_js' ], 5 );
+		add_action( 'wp_footer', [ $this, 'load_render_js' ] );
+	}
+
+
+	/**
+	 * Loads payment gateway render script if checkout page or pay page.
+	 *
+	 * @since 5.10.5
+	 */
+	public function load_render_js() {
```

Comment:
I'd rename this method with `maybe_render_js` since it's not loading anything

you could also hook both actions to the same callback and provide a conditional in case the current action is `wp_footer`

Reply:
Do you think it's a good idea to change the method name? we have to consider that it will result updating the implementation in all gateway plugins for the new name ... how about after renaming the method, check if the old method exists and call it to introduce a layer of compatibility, ex:
```php
if (method_exists($this, 'load_render_js')) {
	$this->load_render_js();
}
```

Reply:
did `load_render_js` existed in the gateways? I didn't think so